Работа над дипломным проектом 3. UI тестирование

mchvell opened the pull request:
Добавлены:
- Тесты
- POM
- Генератор данных
- Отчет

Commit:
Работа над дипломным проектом 3. UI тестирование

Добавлены:
- Тесты
- POM
- Генератор данных
- Отчет

AlexKitIt commented on data/user_generator.py:
Нужно исправить: в data пакетах хранятся статичные тестовые данные. Методы с генерацией данных должны быть в helpers, а логические методы стоит определить в api_modules

mchvell replied:
сделал, как понял:
- сделал пакет api_modules в котором сделал файл api_modules – в нем работаем с апи
- в user_generator оставил только методы, которые создают данные

AlexKitIt commented on pages/auth_page.py:
```diff
@@ -0,0 +1,30 @@
+    def fill_auth_form(self, **kwargs):
+        for field, value in kwargs.items():
+            locator = self.input_locators.get(field)
+            self.find_element(locator, 10).send_keys(value)
```
Можно лучше: такие методы тоже стоит описать в Base как и клик

mchvell replied:
улучшил!

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,29 @@
+
+    @allure.step("Найти элемент")
+    def find_element(self, locator, time=15):
+        return WebDriverWait(self.driver, time).until(EC.element_to_be_clickable(locator),
+                                                      message=f"Элемент {locator} не найден")
+
+    @allure.step("Найти элементы")
+    def find_elements(self, locator, time=15):
+        return WebDriverWait(self.driver, time).until(EC.presence_of_all_elements_located(locator),
+                                                      message=f"Элемент {locator} не найден")
+
+    @allure.step("Перейти на по урлу")
+    def go_to_site(self):
+        return self.driver.get(self.base_url)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver и waitы необходимо вынести в base, там описать методы взаимодействия с ними и оттуда за счет наследования классов страниц от base переиспользовать

mchvell replied:
вроде бы исправил, также вынес клики, различные вейтинги для элементов, сильно меньше кода не стало, тк костыльные методы для получения заказов сегодня тугие(

AlexKitIt commented on tests/test_constructor_on_main.py:
```diff
@@ -0,0 +1,51 @@
+class TestMainPageConstructor:
+    @allure.description("Проверяем, что при нажатии на игридиент открывается модальное окно")
+    @pytest.mark.parametrize("browser", ["chrome", "firefox"], indirect=True)
+    def test_click_on_ingredient(self, browser):
```
Нужно исправить: здесь и далее: над каждым тестовым методом  ставится аннотация allure.title

mchvell replied:
description заменен на title, те исправлено

AlexKitIt commented on tests/test_nav_bar.py:
```diff
@@ -0,0 +1,33 @@
+
+        nav_bar = NavBarHelper(browser)
+        nav_bar.switch_tab("order_feed")
+        url = browser.current_url
```
Нужно исправить: здесь и далее: прямого обращения к browser, вызова его property, execute_script не нем и тд из тестов быть не должно. Для этого в base пишутся соответсвующие методы и тут вызываются на объектах конкретных страниц

mchvell replied:
Исправлено

Commit:
Работа над Дипломным проектом. Правки после ревью 1

- Вынес методы в base page, чтобы использовать их в хелперах
- Обновил отчет
- Вынес модуль работы с апи в отдельный пакет
- Улучшил фикстуру

ps. меня пугает что в папке с отчетом ТАК много файлов
